### ISSUE TITLE
Forbid assigning `lambda` to object properties

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1733]
# Rule request

## Thesis

We should forbid code like: 

```python
class Example(object):
    def __init__(self):
        self.callback = lambda arg: arg + 1
```

## Reasoning

This will be very tricky to understand for users: `example.callback(1)` will look like a method, but it is not a method.

**Comment by abhi-krish** [comment]
Hi! I'd like to work on this.

**Comment by sobolevn** [comment]
Awesome! Thanks a lot!

**Comment by abhi-krish** [comment]
Hey @sobolevn! This is my first time contributing and am not sure where I should start with this problem. Would you be able to point me in the right direction?

**Comment by sobolevn** [comment]
Sure!

Start with reading this file: https://github.com/wemake-services/wemake-python-styleguide/blob/master/CONTRIBUTING.md
Then look at: https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/visitors/ast/classes.py

**Comment by willtkwon** [comment]
Hi, can I work on this task?

**Comment by sobolevn** [comment]
@willtkwon Please feel free! And thank you 🙂

**Comment by AlexandrKhabarov** [comment]
Is it correct that this check should find situations like this?
```python
create_object.attr = lambda: print('Lambda!')
```
I mean that it should not distinguish lambda assignemnts inside __init__ or anywehere.

**Comment by sobolevn** [comment]
We only treat `self`, `cls`, and `mcs` as self-like objects. So, we only check them. See https://github.com/wemake-services/wemake-python-styleguide/blob/master/wemake_python_styleguide/constants.py#L150
